frontend/packages/schema-model/src/lib/mutations/ui-schema-reducers.ts

```diff
@@ -268,13 +269,34 @@ export const setPropertyName: UiSchemaReducer<SetPropertyNameArgs> =
     return renameNodePointer(newSchema, nodeToRename.pointer, newPointer);
   };
 
-export const toggleArrayField: UiSchemaReducer<string> =
-  (uiSchema, pointer) => {
-    const newSchema = deepCopy(uiSchema);
-    const node = getNodeByPointer(newSchema, pointer);
-    node.isArray = !node.isArray;
-    return newSchema;
-  };
+
+// update the "children" pointers 
+//include items in the pointer path if isArray is toggled on, or
+// remove items from the pointer path if isArray is toggled off. 
+// if isArray is toggled on, add items to the pointer path
+export const toggleArrayField: UiSchemaReducer<string> = (uiSchema, pointer) => {
+  const newSchema = deepCopy(uiSchema);
+  const node = getNodeByPointer(newSchema, pointer);
+  node.isArray = !node.isArray;
+
+  if (node.isArray) {
+    node.children.forEach((child) => {
+      const childNode = getNodeByPointer(newSchema, child);
+      childNode.pointer.replace(childNode.pointer, makePointer(
+        childNode.pointer, Keyword.Items));
```

Comment:
There are a few issues with this replace-code.

Firstly, the arguments to the replace function mean you basically replace the whole child pointer (`childNode.pointer`) with the child pointer followed by `/items`. You might as well have written `childNode.pointer = makePointer(childNode.pointer, Keyword.Items);`, it would result in the same string.
So if `childNode.pointer` was `#/properties/parentNode/properties/childNode`, this replace-statement would result in `#/properties/parentNode/properties/childNode/items`.

What we want to end up with is `#/properties/parentNode/items/properties/childNode`.

The parentNode pointer in this case would be `#/properties/parentNode`, which is what we want to replace with `#/properties/parentNode/items`.

The parentNode pointer in this function is contained in the `pointer` variable, which you should use in the replace-statement instead of `childNode.pointer`.

Secondly, the `replace`-function does not actually change the string it is running against, it returns a new string. So in this code, you are not actually changing the `childNode.pointer` at all.

I have created a code sandbox where you can take a look at some different replace statements and how they affect the result (open the console at the bottom of the preview pane on the right to see the output): https://codesandbox.io/s/admiring-galois-gk9gp6?file=/src/index.ts

I suggest you try and play around a bit with this until you get a feel for how the replace method works and which parameters you should pass in.

Once you have done that I would suggest you add some console.log-statements in this code here while you are working, so that you can log the values that you generate and verify that they are as you expect. This makes it much easier to understand what is wrong if the code you write doesn't to as you expect.